geode-pulse/src/main/java/org/apache/geode/tools/pulse/internal/security/LogoutHandler.java

```diff
@@ -30,24 +33,31 @@
 
 /**
  * Handler is used to close jmx connection maintained at user-level
- *
  */
-public class LogoutHandler extends SimpleUrlLogoutSuccessHandler implements LogoutSuccessHandler {
+public class LogoutHandler extends SimpleUrlLogoutSuccessHandler implements LogoutSuccessHandler,
```

Comment:
I believe you don't need to implement `LogoutSuccessHandler` since `SimpleUrlLogoutSuccessHandler` already implements that

you don't need to implement `ContextAware` either, Spring should take care of auto-wiring for you if you declare
<code>
@Autowired
  private Repository repository;
<code>

Reply:
I've removed the unnecessary `implements LogoutSuccessHandler`.

Removing `ContextAware` is trickier. Unfortunately, simply autowiring the repository this way causes a circular dependency that Spring can't resolve.

In my logout spike, I have fixed this by untangling the circular dependency. So here are the options I can think of:

1. Move the dependency-untangling code from my logout spike into this PR. This involves changes to 8 or 9 files.
2. Leave `LogoutHandler` implementing `ContextAware` for now, and fix it in a later PR.

Which would you prefer?

Reply:
we can leave it for now